pkg/distributor/validate.go

```diff
@@ -455,7 +465,11 @@ func validateLabels(m *sampleValidationMetrics, cfg labelValidationConfig, userI
 		} else if len(l.Value) > maxLabelValueLength {
 			cat.IncrementDiscardedSamples(ls, 1, reasonLabelValueTooLong, ts)
 			m.labelValueTooLong.WithLabelValues(userID, group).Inc()
-			return fmt.Errorf(labelValueTooLongMsgFormat, l.Name, l.Value, mimirpb.FromLabelAdaptersToString(ls))
+			return LabelValueTooLongError{
+				Label:  labels.Label{Name: strings.Clone(l.Name), Value: strings.Clone(l.Value)},
+				Series: mimirpb.FromLabelAdaptersToString(ls),
```

Comment:
What happens if the label string is truly enormous, like 100MB?
(it was doing the generation before, but now you hang on to that string for longer)